boxes/generators/dicebox.py

```diff
@@ -0,0 +1,123 @@
+            "--lidheight",  action="store", type=float, default=18,
+            help="height of lid in mm")
+        self.argparser.add_argument(
+            "--hex_hole_corner_radius",  action="store", type=float, default=5,
+            help="The corner radius of the hexagonal dice holes, in mm")
+        self.argparser.add_argument(
+            "--magnet_diameter",  action="store", type=float, default=6,
+            help="The diameter of magnets for holding the box closed, in mm")
+
+    def diceCB(self):
+        t = self.thickness
+        xi = self.x - 2 * t
+        yi = self.y - 2 * t
+        xc = xi / 2
+        yc = yi / 2
+        cr = self.hex_hole_corner_radius
+
+        # -4*t because there are four gaps across:
+        #   2 between the outer holes and the finger joints
+        #   2 between the outer holes and the center hole
+        # /6 because there are 6 apothems across, 2 for each hexagon
+        apothem = (min(xi, yi) - 4 * t) / 6
+        r = apothem * 2 / math.sqrt(3)
+
+        # dice
+        centers = [[xc, yc]]  # start with one in the center
+        polar_r = 2 * apothem + t  # the full width of a hexagon, plus a gap of t width
+        for i in range(6):
+            theta = i * math.pi / 3 + math.pi / 6  # 60 degrees each step, plus a 30 degree offset
+            centers.append(
+                [
+                    xc + polar_r * math.cos(theta),
+                    yc + polar_r * math.sin(theta),
+                ]
+            )
+        for center in centers:
+            self.regularPolygonHole(x=center[0], y=center[1], n=6, r=r, corner_radius=cr)
+
+        # magnets
+        d = self.magnet_diameter
+        self.hole(t+d/2, t+d/2, d=d)
+        self.hole(t+d/2, self.x-3*t-d/2, d=d)
+
+    def render(self):
+        x, y, h, hl = self.x, self.y, self.h, self.lidheight
+
+        if self.outside:
+            x = self.adjustSize(x)
+            y = self.adjustSize(y)
+            h = self.adjustSize(h)
+            hl = self.adjustSize(hl)
+
+        t = self.thickness
+
+        hy = self.edges["O"].startwidth()
+        hy2 = self.edges["P"].startwidth()
+
+        e1 = edges.CompoundEdge(self, "eF", (hy-t, h-hy+t))
+        e2 = edges.CompoundEdge(self, "Fe", (h-hy+t, hy-t))
+        e_back = ("F", e1, "F", e2)
+
+        p = self.edges["o"].settings.pin_height
+        e_inner_1 = edges.CompoundEdge(self, "fe", (y-p, p))
+        e_inner_2 = edges.CompoundEdge(self, "ef", (p, y-p))
+        e_inner_topbot = (e_inner_1, "f", e_inner_2, "f")
+
+        self.ctx.save()
+
+        self.rectangularWall(x, y, "ffff", move="up")
+        self.rectangularWall(x, y, "ffff", move="up")
+        self.rectangularWall(x, h, "FFFF", ignore_widths=[1,2,5,6], move="up")
+        self.rectangularWall(x, h, e_back, move="up")
+        self.rectangularWall(x, hl, "FFFF", ignore_widths=[1,2,5,6], move="up")
+        self.rectangularWall(x, hl-hy2+t, "FFqF", move="up")
+
+        self.ctx.restore()
```

Comment:
This one is still the wrong way round and only moves y to the side when it needs to offset the x column.

Reply:
@florianfesti ohhh I get it; I did not really understand the "only" moves before. Fixed, thanks.

Reply:
Yeah, it isn't that obvious but very convenient once one understands how to use it...